bookkeeper-server/src/main/java/org/apache/bookkeeper/client/RegionAwareEnsemblePlacementPolicy.java

```diff
@@ -644,12 +645,75 @@ public final DistributionSchedule.WriteSet reorderReadLACSequence(
     @Override
     public PlacementPolicyAdherence isEnsembleAdheringToPlacementPolicy(List<BookieId> ensembleList,
             int writeQuorumSize, int ackQuorumSize) {
-        /**
-         * TODO: have to implement actual logic for this method for
-         * RegionAwareEnsemblePlacementPolicy. For now return true value.
-         *
-         * - https://github.com/apache/bookkeeper/issues/1898
-         */
+        if (CollectionUtils.isEmpty(ensembleList)) {
+            return PlacementPolicyAdherence.FAIL;
+        }
+
+        int effectiveMinRegionsForDurability = disableDurabilityFeature.isAvailable() ? 1 : minRegionsForDurability;
+
+        int ensembleSize = ensembleList.size();
+        Map<String, Set<BookieId>> regionsInQuorum = new HashMap<>();
+        BookieId bookie;
+        for (int i = 0; i < ensembleList.size(); i++) {
+            regionsInQuorum.clear();
+            for (int j = 0; j < writeQuorumSize; j++) {
+                bookie = ensembleList.get((i + j) % ensembleSize);
+                if (knownBookies.containsKey(bookie)) {
+                    String region = getLocalRegion(knownBookies.get(bookie));
+                    if (regionsInQuorum.containsKey(region)) {
+                        regionsInQuorum.get(region).add(bookie);
+                    } else {
+                        Set<BookieId> bookieSet = new HashSet<>();
+                        bookieSet.add(bookie);
+                        regionsInQuorum.put(region, bookieSet);
+                    }
+                } else if (LOG.isDebugEnabled()) {
+                    LOG.debug("bookie {} is not in the list of knownBookies", bookie);
+                }
+            }
+
+            if (regionsInQuorum.isEmpty()) {
+                return PlacementPolicyAdherence.FAIL;
+            }
+
+            if (regionsInQuorum.size() < 2) {
```

Comment:
I have a question: why judge the `regionsInQuorum.size() < 2`?

Reply:
The purpose of doing this here is to align the implement in `org.apache.bookkeeper.client.RegionAwareEnsemblePlacementPolicy#newEnsemble`:
https://github.com/apache/bookkeeper/blob/185fef6df92713a92f1531a4044f75ffebcc6e93/bookkeeper-server/src/main/java/org/apache/bookkeeper/client/RegionAwareEnsemblePlacementPolicy.java#L356-L366

When there is only one available region, fall back to `RackAwareEnsemblePlacement`.

Reply:
It's a little weird. 
Based on: writeQuorumSize = 3.

If regionsInQuorum < 2, the result may be `MEETS_STRICT`.

If regionsInQuorum = 2, the result is `FAIL`.

Reply:
Yes, a little weird.

Should we just remove the `numRegionsAvailable < 2` judge in `isEnsembleAdheringToPlacementPolicy`? May I have your suggestion?

Reply:
The PlacementPolicyAdherence enum has three values `FAIL(1), MEETS_SOFT(3), MEETS_STRICT(5)`.
If there are enough different region, the result could be `MEETS_STRICT`. If there aren't enough different region, the each RackAwarePLacementPolicy `MEETS_STRICT`, the result could be `MEETS_SOFT`

Reply:
Updated.